From e10b7035f599eb6d42780cf77336f4ed35ee92be Mon Sep 17 00:00:00 2001
From: "Rajat Soni(Temp)" <quic_rajson@quicinc.com>
Date: Wed, 13 Jul 2022 16:59:23 +0530
Subject: [PATCH] ath12k: move warn prints to dbg when srng is full

Srng full failures in ath12k, it is expected behavior
if ingress traffic is unlimited.
Move ath12k_warn prints to ath12k_dbg if ret == -ENOMEM.

Signed-off-by: Rajat Soni(Temp) <quic_rajson@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/mac.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/mac.c b/drivers/net/wireless/ath/ath12k/mac.c
index 1d16b6b..5655f3c 100644
--- a/drivers/net/wireless/ath/ath12k/mac.c
+++ b/drivers/net/wireless/ath/ath12k/mac.c
@@ -6505,8 +6505,13 @@ static void ath12k_mac_op_tx(struct ieee80211_hw *hw,
 		arsta = (struct ath12k_sta *)control->sta->drv_priv;
 
 	ret = ath12k_dp_tx(ar, arvif, arsta, skb);
-	if (ret) {
-		ath12k_warn(ar->ab, "failed to transmit frame %d\n", ret);
+	if (unlikely(ret)) {
+		if (ret == -ENOMEM)
+			ath12k_dbg(ar->ab, ATH12K_DBG_MAC, "failed to transmit frame %d\n",
+				   ret);
+		else
+			ath12k_warn(ar->ab, "failed to transmit frame %d\n",
+				    ret);
 		ieee80211_free_txskb(ar->hw, skb);
 	}
 }
-- 
2.17.1

